lib/task_listener.js

```diff
@@ -322,43 +322,40 @@ export default class TaskListener extends EventEmitter {
    * directly from the TC queue service, then calling runTask with the
    * resulting task set.
    */
-  async coalesceClaim(claim) {
+  async applySuperseding(claim) {
     let task = claim.task;
     let taskId = claim.status.taskId;
 
     try {
-      // if the task is not coalescible, then the taskset is just the one claim
-      if (!task.payload.coalescer) {
-        this.runtime.log('not coalescing', {taskId, message: 'no coalescer in payload'});
+      // if the task is not set up to supersede anything, then the taskset is just the one claim
+      if (!task.payload.supersederUrl) {
+        this.runtime.log('not superseding', {taskId, message: 'no supersederUrl in payload'});
         return [claim];
       }
 
-      let routes = task.routes;
-      let routePrefix = task.payload.coalescer.routePrefix + ".";
-      routes = routes.filter(r => r.startsWith(routePrefix));
-      if  (routes.length != 1) {
-        this.runtime.log('not coalescing', {taskId, message: 'no single coalescer route in payload'});
+      let supersederUrl = task.payload.supersederUrl;
+
+      let tasks = (await this.fetchSupersedingTasks(supersederUrl, taskId));
+      if (!tasks || tasks.length == 0 || tasks[0] === taskId) {
+        this.runtime.log('not superseding', {taskId, supersederUrl,
+                           message: 'no tasks supersede this one'});
         return [claim];
       }
 
-      let coalescingKey = routes[0].substr(routePrefix.length);
-      let coalescerUrl = task.payload.coalescer.url + coalescingKey;
-
-      let tasks = (await this.fetchCoalescerTasks(coalescerUrl))[coalescingKey];
-      if (tasks.length == 0) {
-        this.runtime.log('not coalescing', {taskId, coalescerUrl,
-                           message: 'no tasks to coalese with'});
+      // if the returned list does not contain the initial taskId, then ignore
+      // the request; this is an invalid response from the superseder
+      let taskIdIndex = tasks.indexOf(taskId);
```

Comment:
if you don't care about the array position, you could do:
`if (!tasks.includes(taskId)) {`